crates/ty_server/src/session.rs

```diff
@@ -414,11 +439,31 @@ impl Session {
         client: &Client,
     ) {
         assert!(!self.workspaces.all_initialized());
+
+        // These are the options combined from all the global options received by the server for
+        // each workspace via the workspace configuration request.
+        let mut combined_global_options: Option<GlobalOptions> = None;
+
         for (url, options) in workspace_settings {
             tracing::debug!("Initializing workspace `{url}`");
 
-            let settings = options.into_settings();
-            let Some((root, workspace)) = self.workspaces.initialize(&url, settings) else {
+            // Combine the global options specified during initialization with the
+            // workspace-specific options to create the final workspace options.
+            let ClientOptions { global, workspace } = self
+                .initialization_options
+                .options
+                .clone()
+                .combine(options.clone());
+
+            combined_global_options = if let Some(global_options) = combined_global_options.take() {
+                Some(global_options.combine(global))
+            } else {
+                Some(global)
+            };
```

Comment:
Can't you call `combine` directly on `combined_global_options` (`Combine` is implemented for `Option<T>` where `T: Combine`)